### PR TITLE
BAAS-24170: Prevent setting rate limiter to nil

### PR DESCRIPTION
**Pull request by Calvinnix**
This change just means we can't override the limiter with a nil limiter.

This works now because we are defaulting the limiter [here](https://github.com/Calvinnix/goja/blob/9b379545217111b76b9286649a58ac61685ecd31/runtime.go#L508).

### CODE REVIEW

File: limiter.go [hunk 1]
```diff
@@ -6,11 +6,10 @@ import (
 
 // SetRateLimiter sets the rate limiter
 func (self *Runtime) SetRateLimiter(limiter *rate.Limiter) {
-	self.limiter = limiter
 	if limiter == nil {
 		return
```
Comment (arahmanan) [inline limiter.go:10]: [opt/q] would it be more intuitive to set `self.limiter` to `rate.NewLimiter(rate.Inf, maxInt)` when `limiter` is nil? That way if the default value is changed to something else, this won't still be a noop. I don't have strong feelings about this at all, so feel free to ignore.

File: limiter.go [hunk 2]
```diff
@@ -6,11 +6,10 @@ import (
 
 // SetRateLimiter sets the rate limiter
 func (self *Runtime) SetRateLimiter(limiter *rate.Limiter) {
```
Comment (arahmanan) [inline limiter.go:8]: [opt/q] Is it worth adding a test that confirms that we won't panic with a nil limiter?

**Review by arahmanan — APPROVED** [review]
LGTM! just a couple of optional comments

**Review by andrewzlchen — APPROVED** [review]
looks good to me mod Andy's comment!
